$OpenBSD: patch-src_util_c,v 1.1 2012/05/23 09:16:19 ajacoutot Exp $

Fix a bug which broke DHCPv6/RA with prefix lengths which are not divisible
by 8.

--- src/util.c.orig	Mon May 21 13:55:18 2012
+++ src/util.c	Mon May 21 13:55:53 2012
@@ -330,7 +330,7 @@ int is_same_net6(struct in6_addr *a, struct in6_addr *
     return 0;
 
   if (pfbits == 0 ||
-      (a->s6_addr[pfbytes] >> (8 - pfbits) != b->s6_addr[pfbytes] >> (8 - pfbits)))
+      (a->s6_addr[pfbytes] >> (8 - pfbits) == b->s6_addr[pfbytes] >> (8 - pfbits)))
     return 1;
 
   return 0;
